scripts/vbumper/main.go

```diff
@@ -0,0 +1,166 @@
+	_skipChangelog = flag.Bool("skip-changelog", false, "Skip updating the changelog")
+)
+
+func main() {
+	*_versionDate = time.Now().Format("2006-01-02")
+	flag.Parse()
+
+	if *_version == "" {
+		log.Fatal("Please specify the version to release using --version")
+	}
+	*_version = strings.TrimPrefix(*_version, "v")
+
+	prevVersion, err := updateChanges()
+	if err != nil {
+		log.Fatal("failed to update changelog", err)
+	}
+
+	if err := updateVersion(prevVersion); err != nil {
+		log.Fatal("failed to update version", err)
+	}
+}
+
+func updateVersion(prevVersion string) error {
+	versionBytes, err := ioutil.ReadFile(*_versionFile)
+	if err != nil {
+		return err
+	}
+
+	newContents := insertNewVersion(string(versionBytes), prevVersion, *_version)
+	return ioutil.WriteFile(*_versionFile, []byte(newContents), 0666)
+}
+
+func insertNewVersion(contents, prevVersion, newVersion string) string {
+	// Find the version string in the file
+	versionStart := strings.Index(contents, prevVersion)
+	versionLine := contents[versionStart:]
+	versionEnd := strings.Index(versionLine, `"`) + versionStart
+	return contents[:versionStart] + newVersion + contents[versionEnd:]
+}
+
+func updateChanges() (oldVersion string, _ error) {
+	changelogBytes, err := ioutil.ReadFile(*_changelogFile)
+	if err != nil {
+		return "", err
+	}
+
+	newLog, oldVersion, err := insertNewChangelog(string(changelogBytes))
+	if err != nil {
+		return "", err
+	}
+
+	if *_skipChangelog {
+		return oldVersion, nil
+	}
+
+	return oldVersion, ioutil.WriteFile(*_changelogFile, []byte(newLog), 0666)
+}
+
+func insertNewChangelog(contents string) (string, string, error) {
+	prevVersionHeader := strings.Index(contents, "# ")
+	versionLine := contents[prevVersionHeader:]
+	lastVersionEnd := strings.Index(versionLine, "(")
+	prevVersionTag := strings.TrimSpace(versionLine[1 : lastVersionEnd-1])
+
+	newChanges, err := getNewChangelog(prevVersionTag)
+	if err != nil {
+		return "", "", err
+	}
+
+	// Strip the 'v' prefix.
+	prevVersion := prevVersionTag[1:]
+	newContents := contents[:prevVersionHeader] + newChanges + contents[prevVersionHeader:]
+	return newContents, prevVersion, nil
+}
+
+func getNewChangelog(prevVersion string) (string, error) {
+	changes, err := getChanges(prevVersion)
+	if err != nil {
+		return "", err
+	}
+
+	if len(changes) == 0 {
+		changes = []string{"No changes yet"}
+	}
+
+	buf := &bytes.Buffer{}
+	_changeTmpl.Execute(buf, struct {
+		Version string
+		Date    string
+		Changes []string
+	}{
+		Version: *_version,
+		Date:    *_versionDate,
+		Changes: changes,
+	})
+	return buf.String(), nil
+}
+
+var _changeTmpl = template.Must(template.New("changelog").Parse(
+	`# v{{ .Version }} ({{ .Date }})
+{{ range .Changes }}
+ * {{ . -}}
+{{ end }}
+
+`))
+
+func getChanges(prevVersion string) ([]string, error) {
+	cmd := exec.Command("git", "log", "--oneline", "--no-decorate", prevVersion+"..HEAD")
```

Comment:
Also, you probably want `--no-merges` here as well otherwise your changelog will include "Merge $branch into master" or similar if you ever do a merge commit.